### ISSUE TITLE
Crash when deleting text from start-process command

### ISSUE THREAD

**Issue by sirandyb** [PowerShell/PSReadLine#911]
Environment data
----------------

PS version: 5.1.17763.316
PSReadline version: 2.0.0-beta2
os: 10.0.17763.1 (WinBuild.160101.0800)
PS file version: 10.0.17763.1 (WinBuild.160101.0800)

Steps to reproduce or exception report
--------------------------------------
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 55 Keys:
 s t a r t - p r o c e s s Space $ e n v : w i n d i r \ n o t e p a d . e x e Space | Space o u t - n u l l Home Delete Delete Delete Delete Delete Delete

Exception:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Microsoft.PowerShell.PSConsoleReadLine.GenerateRender(String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.DeleteCharImpl(Int32 qty, Boolean orExit)
   at Microsoft.PowerShell.PSConsoleReadLine.DeleteChar(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------

**Comment by msftrncs** [comment]
@daxian-dbw, have you started on this yet?  I guess I should have seen this when I worked on PR #989 (token bleed), as it would seem the solution to this is to add a `continue` (now available because of the `while` that I added) after the `peek` when the `tokenStack` still has more items.  

Evidently, `process` is one of very few situations in the parser that the parser stops processing due to an error (or unexpected situation), as nothing past ~a pipe, when the pipe follows a `process` keyword~ the next token after `process` gets parsed (including no EOS token).

I'll post a PR shortly.
